python/repositories.bzl

```diff
@@ -1,8 +1,199 @@
+# Copyright 2022 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 """This file contains macros to be called during WORKSPACE evaluation.
 
 For historic reasons, pip_repositories() is defined in //python:pip.bzl.
 """
 
+load("//python/private:toolchains_repo.bzl", "PLATFORMS", "toolchains_repo")
+load("//python/private:versions.bzl", "MINOR_MAPPING", "RELEASE_DATE", "RELEASE_URL", "TOOL_VERSIONS")
+
 def py_repositories():
     # buildifier: disable=print
     print("py_repositories is a no-op and is deprecated. You can remove this from your WORKSPACE file")
+
+########
+# Remaining content of the file is only used to support toolchains.
+########
+_DOC = "Fetch external tools needed for python toolchain."
+_ATTRS = {
+    "integrity": attr.string(mandatory = True),
```

Comment:
Why is this using `integrity` and not `sha256`, which I feel is far more common?

Reply:
Took the opportunity to switch it in https://github.com/bazelbuild/rules_python/pull/618/commits/d7ac042f2ad92670f60e25d4f97fcdef567c7859.